palantir-java-format/src/main/java/com/palantir/javaformat/doc/HasUniqueId.java

```diff
@@ -0,0 +1,14 @@
+package com.palantir.javaformat.doc;
+
+import fj.Ord;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public abstract class HasUniqueId {
```

Comment:
would be nice to tidy up their package structure a little bit so this doens't have to be public!

Reply:
Also worth a little comment saying this is just because we need a _stable_ order and functionaljava uses redblack trees not linkedhashset/maps?